### DISCUSSION TITLE
Upgrading

### DISCUSSION THREAD

**Discussion by isc-rsingh** [intersystems/ipm#323] [Ideas]
- Assuming IPM is namespace-specific, can we provide a system level option that defaults to installing IPM in every namespace which is more aligned with our current setup of a per-namespace IPM version?
- Old ZPM upgrading to new IPM:
  - Modules are still namespace specific.
  - Need upgrade step in IPM.Installer to copy data from old global to new ones.
  - Packages themselves installable by old version of ZPM should still be supported by new version of IPM

**Comment by daimor** [comment]
Anyway, upgrade options should be delivered with the first version of IPM
Do not want to repeat the first version of IRIS issues again.
